application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadLifecycleListener.java

```diff
@@ -0,0 +1,112 @@
+package org.togetherjava.tjbot.features.help;
+
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.events.channel.update.ChannelUpdateAppliedTagsEvent;
+import net.dv8tion.jda.api.events.channel.update.ChannelUpdateArchivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.features.EventReceiver;
+
+import java.time.Instant;
+
+import static org.togetherjava.tjbot.db.generated.tables.HelpThreads.HELP_THREADS;
+
+/**
+ * Listens for help thread events after creation of thread. Updates metadata based on those events
+ * in database.
+ */
+public final class HelpThreadLifecycleListener extends ListenerAdapter implements EventReceiver {
+
+    private final HelpSystemHelper helper;
+    private final Logger logger = LoggerFactory.getLogger(HelpThreadLifecycleListener.class);
+    private final Database database;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param helper to work with the help threads
+     * @param database the database to store help thread metadata in
+     */
+    public HelpThreadLifecycleListener(HelpSystemHelper helper, Database database) {
+        this.helper = helper;
+        this.database = database;
+    }
+
+    @Override
+    public void onChannelUpdateArchived(@NotNull ChannelUpdateArchivedEvent event) {
+        if (!event.getChannelType().isThread()) {
+            return;
+        }
```

Comment:
https://javadoc.io/doc/net.dv8tion/JDA/latest/net/dv8tion/jda/api/events/channel/update/ChannelUpdateArchivedEvent.html

according to the docs ChannelUpdateArchivedEvent is only called for Thread channels so this test is redundant

```suggestion
```